File: common/src/main/java/mods/cybercat/gigeresque/client/entity/model/RunnerAlienEntityModel.java
package mods.cybercat.gigeresque.client.entity.model;

import mod.azure.azurelib.common.api.client.model.DefaultedEntityGeoModel;
import mods.cybercat.gigeresque.Constants;
import mods.cybercat.gigeresque.common.entity.impl.runner.RunnerAlienEntity;
import net.minecraft.client.renderer.RenderType;
import net.minecraft.resources.ResourceLocation;

public class RunnerAlienEntityModel extends DefaultedEntityGeoModel<RunnerAlienEntity> {

    public RunnerAlienEntityModel() {
        super(Constants.modResource("runner_alien/runner_alien"), true);
    }

    @Override
    public RenderType getRenderType(RunnerAlienEntity animatable, ResourceLocation texture) {
        return RenderType.entityTranslucent(getTextureResource(animatable));
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/model/RunnerAlienEntityModel.kt
package com.bvanseg.gigeresque.client.entity.model

import com.bvanseg.gigeresque.client.entity.animation.EntityAnimations
import com.bvanseg.gigeresque.client.entity.texture.EntityTextures
import com.bvanseg.gigeresque.common.entity.impl.RunnerAlienEntity
import net.minecraft.util.Identifier
import software.bernie.geckolib3.core.event.predicate.AnimationEvent
import software.bernie.geckolib3.model.AnimatedGeoModel
import software.bernie.geckolib3.model.provider.data.EntityModelData

/**
 * @author Boston Vanseghi
 */
class RunnerAlienEntityModel : AnimatedGeoModel<RunnerAlienEntity>() {
    override fun getModelLocation(entity: RunnerAlienEntity): Identifier = EntityModels.RUNNER_ALIEN
    override fun getTextureLocation(entity: RunnerAlienEntity): Identifier = EntityTextures.RUNNER_ALIEN
    override fun getAnimationFileLocation(animatable: RunnerAlienEntity): Identifier = EntityAnimations.RUNNER_ALIEN

    override fun setLivingAnimations(entity: RunnerAlienEntity, uniqueID: Int, customPredicate: AnimationEvent<*>) {
        super.setLivingAnimations(entity, uniqueID, customPredicate)
        val neck = animationProcessor.getBone("neck")
        val extraData = customPredicate.getExtraDataOfType(EntityModelData::class.java).firstOrNull() ?: return
        neck.rotationY = extraData.netHeadYaw * (Math.PI.toFloat() / 340f)
    }
}